launcher-service/src/middleware/verifyJWTAndReturnfromPrisma.ts

```diff
@@ -0,0 +1,52 @@
+import { Request, Response, NextFunction } from 'express';
+import jwt from 'jsonwebtoken';
+import createError from 'http-errors';
+import { prisma } from '../db';
+
+export const verifyJWTAndReturnfromPrisma = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const token = req.headers.authorization?.split(' ')[1];
+
+  interface TokenPayload {
+    id: number;
+    role: string;
+    iat: number;
+    exp: number;
+  }
```

Comment:
The below code is generated by AI. Check how it is extends JwtPayload. Please use a format like that. 
```
import jwt, { JwtPayload } from 'jsonwebtoken';

interface CustomJwtPayload extends JwtPayload {
  userId: string;
  username: string;
  email: string;
}

const token = 'your_jwt_token_here';
const secret = 'your_jwt_secret_here';

try {
  const decoded = jwt.verify(token, secret) as CustomJwtPayload;
  console.log(decoded.userId);
  console.log(decoded.username);
  console.log(decoded.email);
} catch (err) {
  console.error(err);
}
```

Also, move your interface out of the function.